coordinator/src/orderbook/routes.rs

```diff
@@ -19,57 +27,139 @@ use tokio::sync::broadcast::Sender;
+        .map_err(|e| AppError::BadRequest(format!("{e:#}")))?;
 
-    Json(order)
+    Ok(Json(order))
 }
 
-#[derive(Deserialize, Serialize)]
+#[derive(Deserialize, Serialize, Clone)]
 pub struct NewOrder {
     pub price: Decimal,
     pub quantity: Decimal,
-    pub maker_id: String,
+    pub trader_id: String,
     pub direction: Direction,
+    pub order_type: OrderType,
+}
+
+pub struct MatchParams {
+    // todo: waiting for https://github.com/get10101/10101/pull/244 to land
+    pub taker_order: Order,
+    pub maker_order: Order,
 }
 
 pub async fn post_order(
     State(state): State<Arc<AppState>>,
     Json(new_order): Json<NewOrder>,
-) -> impl IntoResponse {
-    let mut conn = state.pool.clone().get().unwrap();
-    let order = orderbook::db::orders::insert(&mut conn, new_order).unwrap();
+) -> Result<Json<Order>, AppError> {
+    let mut conn = get_db_connection(&state)?;
+    let order = orderbook::db::orders::insert(&mut conn, new_order.clone()).map_err(|e| {
+        AppError::InternalServerError(format!("Failed to insert new order into db: {e:#}"))
+    })?;
 
-    let sender = state.tx_pricefeed.clone();
-    update_pricefeed(PriceFeedResponseMsg::NewOrder(order.clone()), sender);
+    if new_order.order_type == OrderType::Limit {
+        // we only tell everyone about new limit orders
+        let sender = state.tx_pricefeed.clone();
+        update_pricefeed(PriceFeedResponseMsg::NewOrder(order.clone()), sender);
+    }
 
-    Json(order)
+    let all_orders =
+        orders::all_by_direction_and_type(&mut conn, order.direction.opposite(), OrderType::Limit)
+            .map_err(|e| {
+                AppError::InternalServerError(format!("Failed to load all orders: {e:#}"))
+            })?;
+    let matched_orders = match_order(order.clone(), all_orders)
+        .map_err(|e| AppError::InternalServerError(format!("Failed to match order: {e:#}")))?;
+
+    let authenticated_users = state.authenticated_users.lock().await;
+    for matched_param in matched_orders {
+        let maker_order = matched_param.maker_order;
+        let taker_order = matched_param.taker_order;
+        let maker_public_key = maker_order
+            .trader_id
+            .parse()
+            .expect("To have a valid pub key");
+        let taker_public_key = taker_order
+            .trader_id
+            .parse()
+            .expect("To have a valid pub key");
+
+        match (
+            authenticated_users.get(&maker_public_key),
+            authenticated_users.get(&taker_public_key),
+        ) {
+            (Some(maker_sender), Some(taker_sender)) => {
+                maker_sender
+                    .send(PriceFeedResponseMsg::Match)
+                    .await
+                    .unwrap();
+
+                taker_sender
+                    .send(PriceFeedResponseMsg::Match)
+                    .await
+                    .unwrap();
+            }
+            (Some(_), None) => {
+                tracing::error!(
```

Comment:
```suggestion
                tracing::debug!(
```

Reply:
Or is this *actually unexpected*? I think this is something that can and will happen, so I don't think it's an error.

Reply:
I know that clients can dissapear and come back online but I still think this is an error message here.